src/env.h

```diff
@@ -1311,6 +1314,9 @@ class Environment : public MemoryRetainer {
   std::vector<std::string> argv_;
   std::string exec_path_;
 
+  bool is_processing_heap_limit_callback_ = false;
+  int64_t heap_limit_snapshot_taken_ = 0;
```

Comment:
Why int64_t here? Are we really going to be taking that many snapshots?

Reply:
`int64_t` is the only signed integer type currently supported by the options parser

Reply:
This doesn't look like it's an option tho? Looks like a counter that is incremented for each snapshot.

Reply:
But it is compared to an option (which is `int64_t`), so we would either need to cast or we would get compiler warnings because of mismatched types.

Reply:
This counter makes sure that we don't take more snapshots than the number specified by --heapsnapshot-near-heap-limit. See the last test case in test-heapsnapshot-near-heap-limit.js  (I have not written any docs for this since I want to figure out what exactly should be configurable by users before bring it out of WIP)

Reply:
hmm ok. I'm not sure that's the best way to limit it but don't have a better suggestion right now. 

One question I would have on this is: if an application is running and the snapshot limit is triggered, how does the user know to go looking for it? It's good that you have the Debug statements in there already, but there does not appear to be any other signal to the user that the file was generated and it's something they could easily miss. Once the heapsnapshot limit is reached, the mechanism switches itself off without any notification or indication to the user. Perhaps we should make this a bit noisier? When the option is enabled, and a snapshot is taken, a message could always be printed to the terminal, and when the limit is reached we could also get a message?

Reply:
Yes, perhaps we should just switch the "Wrote snapshot to..." log to a direct log to stderr  - V8 already will write the crash reason, the JS stack and (part of) the native stack to stderr before it crashes anyway.

Reply:
I switched the "Wrote snapshot to.." log to a unconditioonal write to stderr.